src/Promise.php

```diff
@@ -97,6 +99,18 @@ public function cancel()
         $this->call($canceller);
     }
 
+    public function when(callable $onResolved)
+    {
+        $this->done(function ($value) use ($onResolved) {
+            $onResolved(null, $value);
+        }, function ($reason) use ($onResolved) {
+            $onResolved(
+                UnhandledRejectionException::nullOrResolve($reason),
+                null
+            );
+        });
```

Comment:
Both `$onResolved` calls should be wrapped with `try` / `catch` blocks and be forwarded to the error handler, currently the one of `Loop`, once the PR is merged, to the promise error handler: https://github.com/async-interop/promise/pull/19